services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/NeutronRouterServiceImpl.java

```diff
@@ -177,6 +179,18 @@ else if (portId == null && subnetId != null) {
         }
         subnet.setAttachedRouterId(routerId);
 
+        List<String> gwPorts = router.getGatewayPorts();
+        if (gwPorts == null || gwPorts.size() == 0) {
+            router.setOwner(subnet.getVpcId());
+        } else {
+            for (String port : gwPorts) {
+                SubnetsWebJson subnet_o = this.routerToSubnetService.getSubnetsByPortId(router.getProjectId(), port);
+                if (!subnet_o.getSubnets().get(0).getVpcId().equals(subnet.getVpcId())) {
```

Comment:
What happens if they are equal?

Reply:
@xieus They should be equal, otherwise a warning message should be issued preventing user to attach gateway to the router.